test/karma/directives/routeHref.test.js

```diff
@@ -105,31 +105,28 @@ describe('routeHref directive', function() {
       RouteProvider.setHtml5Mode(false);
     });
 
-    inject(function($rootScope, $compile, $window, $location, $timeout) {
+    inject(function($rootScope, $compile, $window, $location) {
       let element = $compile('<a route-href="hashUrlWriter()">Link</a>')($rootScope.$new());
       $rootScope.$digest();
 
-      // spyOn($location, 'url');
-      spyOn($window, 'open').and.callThrough();
-
-      let mockEvent = {
-        type: 'click',
-        metaKey: true
-      };
-
-      element.triggerHandler('click', mockEvent);
-      // element.triggerHandler(mockEvent);
-      // element.click(e => mockEvent = e);
-      element.click();
+      $window.location.origin = 'http://localhost';
+      spyOn($window, 'open');
+      spyOn($location, 'url');
 
-      // $timeout.verifyNoPendingTasks();
-      $timeout.flush()
+      let event = jasmine.createSpyObj('clickEvent', {
+        'open': _.noop(),
+        'preventDefault': _.noop()
+      });
+      event.type = 'click';
+      event.metaKey = true;
 
-      expect($window.open, '$window.open should be called').toHaveBeenCalled();
+      element.triggerHandler(event);
+      $rootScope.$digest();
 
-      // expect($window.open, '$window.open should be called').toHaveBeenCalledWith('_blank');
-      // expect($location.url, '$location.url should not be called').not.toHaveBeenCalled();
-      // expect(event.isDefaultPrevented(), 'should prevent event default').toBe(true);
+      const expectedNewWindowUrl = $window.location.origin + '/#/contacts/update/1';
```

Comment:
could extend this test further to test a range of URL's, probably overkill at the moment as this would require the other tests above to be refactored too.